src/custom-element.js

```diff
@@ -1129,10 +1323,23 @@ function createBaseCustomElementClass(win) {
      * @return {!Promise<!./base-element.BaseElement>}
      */
     getImpl(waitForBuild = true) {
-      const waitFor = waitForBuild ? this.whenBuilt() : this.whenUpgraded();
+      const waitFor = waitForBuild ? this.build() : this.createImpl_();
       return waitFor.then(() => this.impl_);
     }
 
+    /**
+     * @return {!Promise<!./base-element.BaseElement>}
+     * @private
+     */
+    createImpl_() {
+      return this.signals_
+        .whenSignal(CommonSignals.READY_TO_UPGRADE)
```

Comment:
I have a naive question. What is the `READY_TO_UPGRADE` signal for? Which piece of code subscribes to the signal. Can we just delete it entirely?

Reply:
It's kind of important for now. This signal says that `implClass` is available and now can be created. When it's created depends on the `build` and other APIs. There's also a slight difference between how V1 and V2 elements work.